File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    answer = ''
    temp = ''
    
    for i in range(1, len(food)):
        food[i] //= 2        # 어차피 몫을 구하니 홀수도 상관 없음
        temp += str(i) * food[i]
        
    answer = temp + '0' + temp[::-1]
    
    return answer

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.swift
import Foundation

func solution(_ food:[Int]) -> String {
    var result = ""
    
    for (idx, value) in food.enumerated() {
        guard idx != 0 else { continue }
        result += String(repeating: String(idx), count: value / 2)
    }
    
    return result + "0" + result.sorted(by: >)
}